yb-voyager/src/tgtdb/postgres.go

```diff
@@ -568,17 +568,25 @@ func (pg *TargetPostgreSQL) ExecuteBatch(migrationUUID uuid.UUID, batch *EventBa
 			}
 		}
 
-		br := conn.SendBatch(ctx, &ybBatch)
+		br := tx.SendBatch(ctx, &ybBatch)
+		closeBatch := func() error {
+			if closeErr := br.Close(); closeErr != nil {
+				log.Errorf("error closing batch: %v", closeErr)
+				return closeErr
+			}
+			return nil
+		}
 		for i := 0; i < len(batch.Events); i++ {
 			_, err := br.Exec()
 			if err != nil {
 				log.Errorf("error executing stmt for event with vsn(%d): %v", batch.Events[i].Vsn, err)
+				closeBatch()
```

Comment:
it will be safer to just log the `closeBatch()` error here if its not nil.

Reply:
closeBatch already logs the error 👍